src/opentimelineio/clip.cpp

```diff
@@ -19,33 +23,117 @@ Clip::~Clip()
 MediaReference*
 Clip::media_reference() const noexcept
 {
-    return _media_reference;
+    auto active = _media_references.find(_active_media_reference_key);
+    return active == _media_references.end() || !active->second
+               ? nullptr
+               : active->second;
+}
+
+Clip::MediaReferences
+Clip::media_references() const noexcept
+{
+    MediaReferences result;
+    for (auto const& m: _media_references)
+    {
+        result.insert(
+            { m.first, dynamic_retainer_cast<MediaReference>(m.second) });
+    }
```

Comment:
To be consistent with `media_reference`'s logic, shouldn't m.second be tested against nullptr and only added if the dynamic_cast is not null?

Reply:
This is the getter code.  The old `media_reference` code was simply doing:
```
return _media_reference;
```
So it could also return a nullptr.  

It might look like we're setting it, since we're copying into a MediaReference map, but the reason for that is because we can't rely on the return value being implicitly converted from a `Retainer<MediaReference>` to a MediaReference* anymore -- we need to do it explicitly by copying into the returned map.

Reply:
That said, it shouldn't really ever be nullptr since the setter is converting nullptr into MissingReference, same as `set_media_reference` was doing.

Reply:
Since it should always be MissingReference, it's good